### PR TITLE
add did selection to connect

### PR DESCRIPTION
**Pull request by shamilovtim**
## Reason for PR

Adds DID selection to web5 connect and displays a mocked PIN for the end of the web5 connect process.

## How it was implemented

- Mock the DWA by encrypting a mocked connection request with xchacha-poly1305 using a 24 bit nonce and a mocked public key for the DWA
- Embed the DWA's public key and the nonce into a QR code
- Scan that QR code with the mobile app
- Decrypt the mocked connection request
- Display a did selection UI to the user
- For each did selected in the mobile app create a mocked permissions grant and send it to the connect server

### To do:
- 🛑 Blocked: currently can't sign real permissions grants (web5 SDK missing permissions)
- ⚠️ Soon: need to smoke test DWN server
- ⚠️ Soon: need a real DWA rather than the mocked DWA

### CODE REVIEW

File: src/features/connect/ConnectQRScanScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  const [isScanned, setIsScanned] = useState(false);
+
+  useEffect(() => {
+    const getBarCodeScannerPermissions = async () => {
+      const { status } = await BarCodeScanner.requestPermissionsAsync();
+      setHasPermission(status === "granted");
+    };
+
+    void getBarCodeScannerPermissions();
+  }, []);
```
Comment (amika-sq) [inline src/features/connect/ConnectQRScanScreen.tsx:24]: Could use `useMount` here!

File: src/features/connect/connect-suite.ts [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+
+const createGrantsForDid = async (
+  selectedDid: string,
+  connectRequest: ConnectRequest
+) => {
+  try {
+    const { privateJwk } = await Secp256k1.generateKeyPair();
+    const authorizationSigner = new PrivateKeySigner({
+      privateJwk,
+      keyId: "did:jank:bob",
```
Comment (amika-sq) [inline src/features/connect/connect-suite.ts:75]: Lol, `did:jank` should be a real thing 🤣

**Review by amika-sq — APPROVED** [review]
LGTM! Thanks for all the really great comments, really helped me grep the mocks. Awesome work!

File: src/features/connect/ConnectProfileSelectScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+  const onPressSubmit = () => {
+    if (decryptedConnectionRequest) {
+      const selectedDids = checkList
+        .filter((box) => box.checked)
+        .map((did) => did.did);
+
+      void ConnectSuite.submitConnection(
+        decryptedConnectionRequest,
+        selectedDids
+      );
```
[the remark this answers is not in the document]
Comment (shamilovtim) [reply src/features/connect/ConnectProfileSelectScreen.tsx:67]: Good point, definitely should await this.

File: src/features/connect/ConnectQRScanScreen.tsx (hunk 1 above)
Comment (shamilovtim) [reply src/features/connect/ConnectQRScanScreen.tsx]: fixed

File: src/features/connect/connect-suite.ts (hunk 2 above)
Comment (shamilovtim) [reply src/features/connect/connect-suite.ts:75]: we should create a did:jank protocol in rust

**Comment by shamilovtim** [comment]
> LGTM! Thanks for all the really great comments, really helped me grep the mocks. Awesome work!

Appreciate the feedback and review. Thanks man!

**Pull request merged**